File: 0647-palindromic-substrings/0647-palindromic-substrings.java
class Solution {
    public int countSubstrings(String s) {
        
        int len = s.length();
        int total_palin_substrings =len;
        for(int i=0;i<len;i++){
            int start=i-1,end =i+1;
            while(start >=0 && end <len){
                if(s.charAt(start)!=s.charAt(end)){
                    break;
                }else{
                    start--;
                    end++;
                    total_palin_substrings++;
                }
            }
        }
        for(int i=0;i<len;i++){
            int start=i-1,end =i;
            while(start >=0 && end <len){
                if(s.charAt(start)!=s.charAt(end)){
                    break;
                }else{
                    start--;
                    end++;
                    total_palin_substrings++;
                }
            }
        }
        return total_palin_substrings;
    }
}

File: 0647-palindromic-substrings/0647-palindromic-substrings.cpp
class Solution {
public:
    int leftIndex=0, rightIndex=0;
    int count = 0;
    int countSubstrings(string s) {
        int n= s.length();
        for(int i=0;i<n;i++){
            findLP(s,i,i);
            findLP(s,i,i+1);
        }
        return count ;
    }
    void findLP(string s, int i, int j){
        int n = s.length();
        while(i>=0 && j<n &&s[i]==s[j]){
            i--;
            j++;
            count++;
        }
        
    }
};